### PR TITLE
[KYUUBI #4522] `use:catalog` should execute before than `use:database`

### LINKED ISSUE #4522: [Bug] The `use:catalog` and `use:database` execution order seems not guaranteed
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

```
    normalizedConf.foreach {
      case ("use:catalog", catalog) =>
        try {
          SparkCatalogShim().setCurrentCatalog(spark, catalog)
        } catch {
          case e if e.getMessage.contains("Cannot find catalog plugin class for catalog") =>
            warn(e.getMessage())
        }
      case ("use:database", database) =>
        try {
          SparkCatalogShim().setCurrentDatabase(spark, database)
        } catch {
          case e
              if database == "default" && e.getMessage != null &&
                e.getMessage.contains("not found") =>
        }
      case (key, value) => setModifiableConfig(key, value)
    }
```

as normalizedConf does not preserve the insertion order

### Affects Version(s)

1.7.0

### Kyuubi Server Log Output

_No response_

### Kyuubi Engine Log Output

_No response_

### Kyuubi Server Configurations

_No response_

### Kyuubi Engine Configurations

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to fix.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by lsm1**
### _Why are the changes needed?_

close #4522 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/session/FlinkSessionImpl.scala [hunk 1]
```diff
@@ -57,16 +57,20 @@ class FlinkSessionImpl(
 
   override def open(): Unit = {
     executor.openSession(handle.identifier.toString)
-    normalizedConf.foreach {
-      case ("use:catalog", catalog) =>
+    normalizedConf.get("use:catalog") match {
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/session/FlinkSessionImpl.scala:60]: How about
```
    val (useCatalogAndDatabaseConf, otherConf) = normalizedConf.partition { case (k, _) =>
      !Set("use:catalog", "use:database").contains(k)
    }
```
?

**Comment by bowenliang123** [comment]
Apply the same changes on Trino session too?

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/session/FlinkSessionImpl.scala [hunk 2]
```diff
@@ -57,7 +57,12 @@ class FlinkSessionImpl(
 
   override def open(): Unit = {
     executor.openSession(handle.identifier.toString)
-    normalizedConf.get("use:catalog") match {
+
+    val (useCatalogAndDatabaseConf, otherConf) = normalizedConf.partition { case (k, _) =>
+      Array("use:catalog", "use:database").contains(k)
+    }
+
+    useCatalogAndDatabaseConf.get("use:catalog") match {
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/session/FlinkSessionImpl.scala:65]:
```suggestion
    useCatalogAndDatabaseConf.get("use:catalog").foreach { catalog =>
    ...
```

**Comment by pan3793** [comment]
Thanks, merged to master/1.7
